### PR TITLE
Real Quantity with Units module comparison operators

### PR DESCRIPTION
**Pull request by hpohekar**

### CODE REVIEW

File: src/ansys/fluent/core/quantity.py [hunk 1]
```diff
@@ -717,16 +717,98 @@ def __eq__(self, other):
...
-    def __ge__(self, other):
-        if (
-            isinstance(other, Quantity)
-            and self.get_dimensions_list() == other.get_dimensions_list()
-            and self._si_unit == other._si_unit
+    def __gt__(self, other):
+        if isinstance(other, Quantity) and (
+            self.get_dimensions_list() != other.get_dimensions_list()
         ):
+            raise ValueError("Incompatible dimensions.")
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/quantity.py:724]: Future refactor: write a single method that checks that the dimensions are compatible and raises an exception if they are not. The method can accept float arguments which are handled as dimensionless quantities. 
```
# throws
def validate_matching_dimensions(self, other):
```

File: src/ansys/fluent/core/quantity.py [hunk 2]
```diff
@@ -717,16 +717,98 @@ def __eq__(self, other):
...
+
+    def __ge__(self, other):
+        if isinstance(other, Quantity) and (
+            self.get_dimensions_list() != other.get_dimensions_list()
+        ):
+            raise ValueError("Incompatible dimensions.")
+        elif isinstance(other, Quantity):
+            return float(self) >= float(other)
+        elif (
+            self.is_dimension_less()
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/quantity.py:744]: I think the name should be `is_dimensionless`.

File: src/ansys/fluent/core/quantity.py [hunk 3]
```diff
@@ -717,16 +717,98 @@ def __eq__(self, other):
...
-        if (
-            isinstance(other, Quantity)
-            and self.get_dimensions_list() == other.get_dimensions_list()
-            and self._si_unit == other._si_unit
+    def __gt__(self, other):
+        if isinstance(other, Quantity) and (
+            self.get_dimensions_list() != other.get_dimensions_list()
         ):
+            raise ValueError("Incompatible dimensions.")
+        elif isinstance(other, Quantity):
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/quantity.py:725]: After we have done *all* the validation at the start of each method, the rest of the method can become something quite simple (we are on safe ground). So, the right-hand side can be converted to float regardless: we don't need to check what it is.

So, each method is something like:
```
validate_matching_dimensions(other)
return float(self) <op> float(other)
```

File: tests/test_quantity.py [hunk 4]
```diff
@@ -520,6 +520,94 @@ def test_power_56():
...
+    with pytest.raises(ValueError) as e_info:
+        assert x >= z
+        assert x >= y
+        assert 5.0 >= r
+
+    with pytest.raises(TypeError) as e_info:
+        assert x >= 5.0
+
+
+def test_gt_59():
```
Comment (dnwillia-work) [inline tests/test_quantity.py:538]: Did you want to test that the operator actually works?  Here you are testing the exceptions so far, which is good...

Comment (hpohekar) [reply tests/test_quantity.py:538]: We have already confirmed the operator's working, so only testing the exceptions here. Thank you.

Comment (dnwillia-work) [reply tests/test_quantity.py:538]: OK, staring at the unit tests I cannot see which unit test confirms they are working.  Am I blind?

Comment (hpohekar) [reply tests/test_quantity.py:538]: I mean Krishnan and I, checked the both left and right equivalents locally, so I have not explicitly added them in these unit tests.

But now I will add them too.

Thank you.

Comment (hpohekar) [reply tests/test_quantity.py:542]: Done. Thank you.

File: src/ansys/fluent/core/quantity.py (hunk 2 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py]: Done. Thank you.

File: src/ansys/fluent/core/quantity.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py:724]: Done. Thank you very much.

File: src/ansys/fluent/core/quantity.py (hunk 3 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py]: Done. Thank you very much.

File: src/ansys/fluent/core/quantity.py [hunk 5]
```diff
@@ -708,7 +708,7 @@ def __eq__(self, other):
         if isinstance(other, Quantity):
```
Comment (h-krishnan) [inline src/ansys/fluent/core/quantity.py:708]: Maybe you can use `validate_matching_dimensions` here too

File: src/ansys/fluent/core/quantity.py [hunk 6]
```diff
@@ -693,7 +693,7 @@ def __sub__(self, other):
             temp_value = self._si_value - other._si_value
```
Comment (h-krishnan) [inline src/ansys/fluent/core/quantity.py:693]: Maybe you can use `validate_matching_dimensions` for `__add__` and `__sub__` as well.

File: src/ansys/fluent/core/quantity.py (hunk 5 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py:700]: Right. Done. Thank you.

File: src/ansys/fluent/core/quantity.py (hunk 6 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py]: Right. Done. Thank you.

File: src/ansys/fluent/core/quantity.py (hunk 5 above)
Comment (hpohekar) [reply src/ansys/fluent/core/quantity.py]: Right. Done. Thank you.

**Review by prmukherj — APPROVED** [review]

**Review by h-krishnan — APPROVED** [review]

**Pull request merged**